src/TLuaInterpreter.cpp

```diff
@@ -5775,13 +5777,104 @@ int TLuaInterpreter::sendTelnetChannel102(lua_State* L)
+    *pItem = nullptr;
+    if (argType == LUA_TNUMBER) {
+        int id = qRound(lua_tonumber(L, index));
+        if (id < 0) {
+            return warnArgumentValue(L, func, QStringLiteral("item ID (%1) invalid, it must be equal or greater than zero").arg(id).toUtf8().constData());
+        }
+        *pItem = host.getActionUnit()->getAction(id);
+        if (!*pItem) {
+            return warnArgumentValue(L, func, QStringLiteral("no button item with ID %1 found").arg(id).toUtf8().constData());
+        }
+        if (!(*pItem)->isPushDownButton()) {
+            *pItem = nullptr;
+            return warnArgumentValue(L, func, QStringLiteral("item ID with %1 is not a push-down button").arg(id).toUtf8().constData());
+        }
+    }
+
+    if (argType == LUA_TSTRING) {
+        QString name = lua_tostring(L, index);
+        if (name.isEmpty()) {
+            return warnArgumentValue(L, func, "item name must not be an empty string");
+        }
+        *pItem = host.getActionUnit()->findAction(name);
+        if (!*pItem) {
+            return warnArgumentValue(L, func, QStringLiteral("no button item with name '%1' found").arg(name).toUtf8().constData());
+        }
+        if (!(*pItem)->isPushDownButton()) {
+            *pItem = nullptr;
+            return warnArgumentValue(L, func, QStringLiteral("item with name '%1' is not a push-down button").arg(name).toUtf8().constData());
+        }
+    }
+
+    if (!*pItem) {
+        // we'll get here if the (index) argument is NOT usable:
+        lua_pushfstring(L, "%s: bad argument #%d type (ID as number or name as string expected, got %s!)",
+                        func, index, luaL_typename(L, index));
+        return lua_error(L); // Does not return!
+    }
+
+    return 0;
+}
+
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getButtonState
 int TLuaInterpreter::getButtonState(lua_State* L)
 {
-    Host& host = getHostFromLua(L);
-    int state;
-    state = host.mpConsole->getButtonState();
-    lua_pushnumber(L, state);
+    auto& host = getHostFromLua(L);
+    if (!lua_gettop(L)) {
+        // The original function only works in the script for a push-down button
+        // and takes no arguments so provide the backwards compatible behaviour
+        // if that is the case:
+        lua_pushnumber(L, host.mpConsole->getButtonState());
+        return 1;
+    }
+
+    TAction* pItem = nullptr;
+    if (int retCount = getTActionFromIdOrName(L, &pItem, 1, __func__)) {
+        return retCount;
+    }
+
+    lua_pushboolean(L, pItem->mButtonState);
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setButtonState
+int TLuaInterpreter::setButtonState(lua_State* L)
+{
+    TAction* pItem = nullptr;
+    if (int retCount = getTActionFromIdOrName(L, &pItem, 1, __func__)) {
+        return retCount;
+    }
+
+    auto checked = getVerifiedBool(L, __func__, 2, "checked");
+
+    if (pItem->mButtonState != checked) {
```

Comment:
:-1: **github-actions**

Both of these are false positives because they do not recognise that this does not actually return execution to the caller function (but instead to the point where `setjmp(...)` was called):
>```cpp
>src/TLuaInterpreter.cpp:5827:9: note: Returning value, which participates in a condition later
>        return lua_error(L); // Does not return!
>````